### ISSUE TITLE
False negative for incompatible override of `(self, *args, **kwargs)` method.

### ISSUE THREAD

**Issue by ethantkoenig** [microsoft/pyright#10848]
**Describe the bug**

`B.foo`'s type signature is not compatible with `A.foo`'s, so I believe this should result in a [reportIncompatibleMethodOverride](https://github.com/microsoft/pylance-release/blob/main/docs/diagnostics/reportIncompatibleMethodOverride.md) error, but it does not. 

Possibly(?) related to https://github.com/microsoft/pyright/issues/10815.

**Code or Screenshots**

```python
from typing import Any

class A:
    def foo(self, *args: Any, **kwargs: Any) -> None:
        pass

class B(A):
    def foo(self) -> None:
        pass
```

Interestingly, if either `args` or `kwargs` (but not both) is removed from `A.foo`'s signature, the error is correctly reported.

**VS Code extension or command-line**

Observed via `pyright` CLI (version 1.1.404), Python version 3.10.18.

**Comment by erictraut** [comment]
Pyright is working correctly here. The signature `*args: Any, **kwargs: Any` is treated specially [in the typing spec](https://typing.python.org/en/latest/spec/callables.html#meaning-of-in-callable). It is the moral equivalent of `Any` for a call signature, so all other signatures are assignable to it, and vice versa. To avoid this false negative, you can use `object` instead of `Any` in the first signature.
